test/test_charm.py

```diff
@@ -89,9 +89,6 @@ def _on_start(self, event: ops.EventBase):
 
         self.assertEqual(charm.started, True)
 
-        with self.assertRaisesRegex(TypeError, "observer methods must now be explicitly provided"):
-            framework.observe(charm.on.start, charm)  # type: ignore
-
```

Comment:
Instead of removing the test, can we just make it `assertRaises(TypeError)`? It'll still raise due to the method test, right?